### DISCUSSION TITLE
Security concerns about a different, but similar tool.

### DISCUSSION THREAD

**Discussion by scottcline** [teslamate-org/teslamate#3914] [General]
https://cybersecuritynews.com/30-tesla-cars-hacked/

Curious if TM is vulnerable to any of the issues mentioned in the above-linked articles?

Love the project and the tool.

**Comment by cwanja** [comment]
So yes, if you run an internet exposed version of TeslaMate and leave the default settings, you could be exposing your data. It's not a 'TeslaMate' issue, it's an individuals configuration issue. Individuals should not be exposing services to the internet if they do not know how to secure it.

**Reply by stauffenberg2020** [reply]
Hi @cwanja , Can you please be bit more clear on "What do you mean by"
- internet exposed version of TeslaMate
- leave the default settings

**Reply by cwanja** [reply]
1. Enabling accessing TeslaMate beyond a local internet. Meaning you install it via a VPS in a data center or a cloud hosted service. For me, I installed it on my home network and it is not accessible outside my home network.
2. If you do allow access via the internet, you should change the default passwords (see the docs).

**Reply by brianmay** [reply]
Yes, you should not allow access to TeslaMate beyond your local network. This is good practise.

But: Trusting your local network 100% is also bad practise. You just need to have one computer on your network be compromised in some fashion, and attackers can still get 100% access to your TeslaMate instance. Hence I always set http authentication via a nginx frontend. As well as limiting access to my local network too.

**Comment by brianmay** [comment]
One thing to note with TeslaMate that doesn't appear to be the case with TeslaLogger, even if an attacker gets access to your database, they cannot get access your tokens unless they have access to your `ENCRYPTION_KEY` also.

But: Having db ports exposed to the Internet, that is just asking for trouble. There normally is no benefit to doing so either. Don't do it.

**Comment by stauffenberg2020** [comment]
I want to ask a few questions to both @brianmay and @cwanja. But I cannot ask such questions publicly. How can I reach out to you guys if you are willing to take that question related to such security settings?

**Reply by cwanja** [reply]
@stauffenberg2020 Best solution is going to be generalize your questions and post them here. It's an open source project and your discussion could be valuable to another user.
